backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/monitoring/VmAnalyzer.java

```diff
@@ -874,15 +874,16 @@ private void proceedDisappearedVm() {
 
             break;
 
+        case RebootInProgress:
         case PoweringDown:
+            if (getVmManager().isColdReboot()) {
```

Comment:
I believe the reason for the `not sure about that..` comment below was that generally the vm shouldn't be in status PoweringDown during cold-reboot. due to the context of this PR, I think I see what happened - cold-reboot is initiated, we call 'shutdown' but the vm is not UP so we think we cannot shut it down and thus power it off instead, and later we override the RebootInProgress state with PoweringDown in the next monitoring cycle, is that the case? if so, how about preventing the vm from switching to PoweringDown state and keep it in RebootInProgress state instead?

Reply:
Actually, the VM's state was `RebootInProgress` in the scenario mentioned in the BZ. I just added it to the `switch` and let `RebootInProgress` and `PoweringDown` share the same logic. I am not sure if we can get into the `PoweringDown` state during the cold reboot and it seems you were not sure at the time of writing the code either :-)

Reply:
ah no, I don't think I wrote that code - but I kept it during a major refactoring I made in this class
yeah, ok, makes sense